decode.go

```diff
@@ -313,7 +308,7 @@ func (p *parser) mapping() *Node {
 type decoder struct {
 	doc     *Node
 	aliases map[*Node]bool
-	terrors []string
+	terrors []UnmarshalError
```

Comment:
I didn‘t check this, maybe we‘d also make this a simple error and use errors.Join instead of append? Can be a separate PR though.

Reply:
I think it's better this way. We can be certain that errors here include location, and it's not just some random `fmt.Errorf` without context.

That said, I'd go further and change `Message string` to `Err error` so exact error (e.g. returned by `UnmarshalYAML` method) can be inspected.